pinot-controller/src/main/java/org/apache/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -298,7 +298,7 @@ private void updateSegmentMetrics(String tableNameWithType, TableConfig tableCon
         (nReplicasIdealMax > 0) ? (nReplicasExternal * 100 / nReplicasIdealMax) : 100);
     _controllerMetrics.setValueOfTableGauge(tableNameWithType, ControllerGauge.SEGMENTS_IN_ERROR_STATE, nErrors);
     _controllerMetrics.setValueOfTableGauge(tableNameWithType, ControllerGauge.PERCENT_SEGMENTS_AVAILABLE,
-        (nSegments > 0) ? (100 - (nOffline * 100 / nSegments)) : 100);
+        (nSegments > 0) ? (long) (100 - (nOffline * 100.0 / nSegments)) : 100);
```

Comment:
Avoid introducing double to avoid the precision issue
```suggestion
        (nSegments > 0) ? (nSegments - nOffline) * 100 / nSegments : 100);
```

Reply:
Good idea!